### PR TITLE
docs: add documentation for Nydus data deduplication

### PR DESCRIPTION
**Pull request by jiangliu**
Add documentation about data deduplication technologies used by Nydus.

Signed-off-by: Jiang Liu <gerry@linux.alibaba.com>

### CODE REVIEW

**Comment by yqleng1987** [comment]
@jiangliu , a new test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@jiangliu , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6982>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Review by imeoer — APPROVED** [review]

File: docs/data-deduplication.md [hunk 1]
```diff
@@ -96,21 +96,72 @@ It uses file-level deduplication do reduce data transferred to/from the HTTP ser
...
+A referenced image may be used when creating an image by `nydus-image create --parent-bootstrap referenced-image-bootstrap ...`.
+When creating the first layer of an image, and a referenced parent image is present, it reconstructs the `ChunkDigestHashTable` from the reference bootstrap file instead of creating an empty one.
+By this way, all data chunks existing in the referenced image will get de-duplicated.
+This is most valuable for software upgrading scenario because there may be many duplicated chunks between the existing version and the new version.
+
+Data dedup among multiple images may help to achieve O1, O2, O3, O4 and O5.
+
+## Data Deduplication on Node
+The above three methods achieve data dedup during image building stage.
+But multi container images may still container duplicated data chunks, for example:
```
Comment (liubogithub) [inline docs/data-deduplication.md:155]: s/still container/still contain/g

File: docs/data-deduplication.md [hunk 2]
```diff
@@ -96,21 +96,72 @@ It uses file-level deduplication do reduce data transferred to/from the HTTP ser
...
+- images containing the same nodejs library.
+- images containing the same java runtime.
+ 
+Even more, a container image may container duplicated data with the host, say running a Ubuntu 16.04 based image on a Ubuntu 16.04 host.
```
Comment (liubogithub) [inline docs/data-deduplication.md:159]: Ditto.

**Comment by yqleng1987** [comment]
@jiangliu , your pull request has been updated. A new test job will be submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@jiangliu , your test job has passed, and no need to test again.

File: docs/data-deduplication.md (hunk 1 above)
Comment (jiangliu) [reply docs/data-deduplication.md]: fixed

File: docs/data-deduplication.md (hunk 2 above)
Comment (jiangliu) [reply docs/data-deduplication.md]: fixed

**Pull request merged**